avi/resource_avi_cloud.go

```diff
@@ -236,10 +243,104 @@ func ResourceAviCloudRead(d *schema.ResourceData, meta interface{}) error {
 	return err
 }
 
+// Verify cloudState is valid vcenter state to configure mgmt IP
+func isValidCloudState(cloudState string) bool {
+	switch cloudState {
+	case
+		"VCENTER_DISCOVERY_COMPLETE_NO_MGMT_NW",
+		"VCENTER_DISCOVERY_COMPLETE",
+		"VCENTER_DISCOVERY_WAITING_DC",
+		"VCENTER_DISCOVERY_ONGOING":
+		return true
+	}
+	return false
+}
+
+// Wait untill vcenter inventory gets completed to configure mgmt IP
+func waitForVcenterState(cloudUuid string, client *clients.AviClient, maxRetry int) error {
+	var robj interface{}
+	var err error
+	path := "api/vimgrvcenterruntime?cloud_uuid=" + cloudUuid
+	i := 0
+	for ; i < maxRetry; i++ {
+		if err = client.AviSession.Get(path, &robj); err == nil {
+			objCount := robj.(map[string]interface{})["count"].(float64)
```

Comment:
shouldn't count be type asserted to int32/int62 instead of floar64.

Reply:
If we try to use int here then we get the error 'interface conversion: interface {} is float64, not int'

Reply:
Yeah turns out Go json unmarshaller uses the javascript spec for converting numbers. And javascript spec says that numbers are 64 bit floats. Ref: https://javascript.info/number